Gordon360/Controllers/RecIM/AffiliationsController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Services.RecIM;
+using Gordon360.Authorization;
+using Gordon360.Static.Names;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    public class AffiliationsController : GordonControllerBase
+    {
+        private readonly IAffiliationService _affiliationService;
+
+        public AffiliationsController(IAffiliationService affiliationService)
+        {
+            _affiliationService = affiliationService;
+        }
+
+        /// <summary>
+        /// Gets all stored affiliations/halls/clubs with associated
+        /// Points/Activities Won : Winning Team
+        /// </summary>
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<AffiliationExtendedViewModel>> GetAllAffiliationDetails()
+        {
+            return Ok(_affiliationService.GetAllAffiliationDetails());
+        }
+
+        /// <summary>
+        /// Gets specific stored affiliation with associated
+        /// Points/Activities Won : Winning Team
+        /// </summary>
+        [HttpGet]
+        [Route("{affiliationName}")]
+        public ActionResult<IEnumerable<AffiliationExtendedViewModel>> GetAffiliationDetailsByName(string affiliationName)
+        {
+            var res = _affiliationService.GetAllAffiliationDetails();
+            return Ok(res.FirstOrDefault(a => a.Name == affiliationName));
+        }
+
+        /// <summary>
+        /// Creates new Affiliation
+        /// </summary>
+        /// <param name="affiliationName"></param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("")]
+        [StateYourBusiness(operation = Operation.ADD, resource = Resource.RECIM_AFFILIATION)]
+        public async Task<ActionResult> PutAffiliation(string affiliationName)
```

Comment:
I was moreso thinking that we don't want duplicate names. So in a sense yes? I don't know what is best practice for handling UX in terms of trying to create an `affiliation` that already exists. Should we throw an error? just ignore it since it makes no difference? I don't mind either way

Reply:
I think Post is better, and you can throw an error. The behavior of doing nothing implies that the user has just successfully created a new Affiliation, when in reality they will be using an existing affiliation that has history, which could be construed as a bug.